### PR TITLE
check for null in trigger states and added execution tests

### PR DESCRIPTION
**Pull request by kaikreuzer**
fixes https://github.com/eclipse/smarthome/issues/5578

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

**Comment by sjka** [comment]
Cool to finally have some tests! 
Did you maybe forget to commit the actual fix?

File: bundles/model/org.eclipse.smarthome.model.rule.tests/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleExecutionTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ */
+public class RuleExecutionTest extends JavaOSGiTest {
+
+    private final static String TESTMODEL_NAME = "testModel.rules";
+    private ModelRepository modelRepository;
+    private ItemRegistry itemRegistry;
+    private EventPublisher eventPublisher;
+    private Event resultEvent;
+
+    private final EventSubscriber eventSubscriber = new EventSubscriber() {
```
Comment (sjka) [inline bundles/model/org.eclipse.smarthome.model.rule.tests/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleExecutionTest.java]: Wouldn't this rather be an excellent candidate for a mock? You could grab the expected event with an `ArgumentCaptor`...

**Comment by kaikreuzer** [comment]
> Did you maybe forget to commit the actual fix?

Haha, seems that got lost when updating the license headers...
I just wanted to proof that the tests fail without the fix ;-)

File: bundles/model/org.eclipse.smarthome.model.rule.tests/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleExecutionTest.java (hunk 1 above)
Comment (kaikreuzer) [reply bundles/model/org.eclipse.smarthome.model.rule.tests/src/org/eclipse/smarthome/model/rule/runtime/internal/engine/RuleExecutionTest.java]: I'd prefer to leave it as it is for now - feel free to refactor it later on, if you think it becomes more readable with a mock.

**Pull request merged**